homeassistant/components/light/tcpbulbs.py

```diff
@@ -0,0 +1,146 @@
+"""
+Support for TCP Connected (Greenwave Reality) lights.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tcpbulbs/
+"""
+import logging
+import voluptuous as vol
+# Import the device class from the component that you want to support
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, Light, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS)
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+# Home Assistant depends on 3rd party packages for API specific code.
+SUPPORTED_FEATURES = (SUPPORT_BRIGHTNESS)
+_LOGGER = logging.getLogger(__name__)
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+
+def pullXML(host, bulbxml):
+    REQUIREMENTS = ['xmltodict']
+    import xmltodict
+    resp = tcpAction(host, 0, 'xml', 0)
+    # Store bulb information as xml file
+    with open(bulbxml, 'wb') as f:
+        f.write(resp.content)
+
+    with open(bulbxml) as fd:
+        doc = xmltodict.parse(fd.read())
+        return doc
+
+def numMap(value, leftMin, leftMax, rightMin, rightMax):

```

Comment:
camelCase naming is not very pythonic, even when some stdlib modules do that. `num_map` would be the last to do here, but a more descriptive name could be useful for future contributors!

Reply:
I'll change all the camelCase. The numMap function is pretty explanatory, it maps a range of numbers to another. I will add a comment though.